src/Browsershot.php

```diff
@@ -716,6 +728,21 @@ protected function createCommand(string $url, string $action, array $options = [
 
         $command['options']['args'] = $this->getOptionArgs();
 
+        if (parse_url($url, PHP_URL_SCHEME) === null) {
+            throw CouldNotTakeBrowsershot::urlHasNoScheme();
+        }
+
+        if (! empty($this->postParams)) {
```

Comment:
You could opt to extract this logic to its own function with a descriptive name.

Reply:
Now it's clean as a whistle. I apologize that my overengineering took so much of your time. =)